test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/PrerequisiteSection.java

```diff
@@ -0,0 +1,401 @@
+        }
+    }
+
+    static PrerequisiteSectionBuilder parseInternal(XContentParser parser) throws IOException {
+        PrerequisiteSectionBuilder builder = new PrerequisiteSectionBuilder();
+        var hasPrerequisiteSection = false;
+        var unknownFieldName = false;
+        ParserUtils.advanceToFieldName(parser);
+        while (unknownFieldName == false) {
+            if ("skip".equals(parser.currentName())) {
+                parseSkipSection(parser, builder);
+                hasPrerequisiteSection = true;
+                maybeAdvanceToNextField(parser);
+            } else if ("requires".equals(parser.currentName())) {
+                parseRequiresSection(parser, builder);
+                hasPrerequisiteSection = true;
+                maybeAdvanceToNextField(parser);
+            } else {
+                unknownFieldName = true;
+            }
+        }
+        if (hasPrerequisiteSection) {
+            builder.validate(parser.getTokenLocation());
+        }
+        return builder;
+    }
+
+    private static void parseFeatureField(String feature, PrerequisiteSectionBuilder builder) {
+        // #31403 introduced YAML test "features" to indicate if the cluster being tested has xpack installed (`xpack`)
+        // or if it does *not* have xpack installed (`no_xpack`). These are not test runner features, so now that we have
+        // "modular" skip criteria let's separate them. Eventually, these should move to their own skip section.
+        if (feature.equals("xpack")) {
+            builder.requireXPack();
+        } else if (feature.equals("no_xpack")) {
+            builder.skipIfXPack();
+        } else {
+            builder.requireYamlRunnerFeature(feature);
+        }
+    }
+
+    // package private for tests
+    static void parseSkipSection(XContentParser parser, PrerequisiteSectionBuilder builder) throws IOException {
+        if (parser.nextToken() != XContentParser.Token.START_OBJECT) {
+            throw new IllegalArgumentException(
+                "Expected ["
+                    + XContentParser.Token.START_OBJECT
+                    + ", found ["
+                    + parser.currentToken()
+                    + "], the skip section is not properly indented"
+            );
+        }
+        String currentFieldName = null;
+        XContentParser.Token token;
+
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            } else if (token.isValue()) {
+                if ("version".equals(currentFieldName)) {
+                    builder.skipIfVersion(parser.text());
+                } else if ("reason".equals(currentFieldName)) {
+                    builder.setSkipReason(parser.text());
+                } else if ("features".equals(currentFieldName)) {
+                    // TODO: legacy - remove
```

Comment:
Should we log a warning here and let people know how to migrate?

Reply:
Added